code/script.js

```diff
@@ -215,111 +220,197 @@ const generateBoard = () => {
+          <button class="filled-button small" id="guessBtn" onclick="guess('${person.name}')">Guess</button>
         </div>
       </div>
     `
   })
 }
+generateBoard(CHARACTERS)
 
 // Randomly select a person from the characters array and set as the value of the variable called secret
-const setSecret = () => {
+const setSecret = (charactersInPlay) => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+
 }
 
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
+
+  winOrLose.style.display = "none";
   charactersInPlay = CHARACTERS
   // What else should happen when we start the game?
+  setSecret(CHARACTERS)
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
   // const value =
-
+  const value = questions.value
   currentQuestion = {
     category: category,
-    // value: value
+    value: value,
   }
 }
 
 // This function should be invoked when you click on 'Find Out' button.
+
 const checkQuestion = () => {
   const { category, value } = currentQuestion
-
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+  // I created a group of console.log to help me check what information it contains and if I can use it in this scope
+  // console.log(secret.name)
+  // console.log(currentQuestion)
+  // console.log(charactersInPlay)
+  // console.log(secret[category])
+  // console.log(value)
 
-  } else if (category === 'accessories' || category === 'other') {
 
+  if (category === 'hair' || category === 'eyes') {
+    let compare = value === secret[category]
+    filterCharacters(compare);
+  }
+
+  if (category === 'accessories' || category === 'other') {
+    let compare = secret[category].includes(value)
+    if (compare === true) {
+      filterCharacters(true)
+    }
+    else if (compare === false) {
+      filterCharacters(false)
+    }
   }
 }
 
-// It'll filter the characters array and redraw the game board.
+
+// It'll filter the characters array and redraw the game board.<
+// Use !== 
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
   // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  if (category === 'hair') {
     if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
```

Comment:
Seems like you need to inject the category as well so that the message shows "black hair" instead of only "back".